### PR TITLE
[JENKINS-60118] - Ensure that UserLanguages telemetry initializer always runs after extensions are augmented

### PR DESCRIPTION
**Pull request by oleg-nenashev**
UserLanguages static initializer sets up a PluginServletFilter. In order to do so, we need an initialized Jenkins instance.  It does not manifest in common Jenkins runs, but it pops up in Jenkinsfile Runner when running in the standalone JAR mode: https://github.com/jenkinsci/jenkinsfile-runner/issues/193

The code is now similar to the ResourceDomainFilter:

```java
    @Initializer(after = InitMilestone.EXTENSIONS_AUGMENTED)
    public static void init() throws ServletException {
        PluginServletFilter.addFilter(new ResourceDomainFilter());
    }
```

See [JENKINS-60118](https://issues.jenkins-ci.org/browse/JENKINS-60118).

### Proposed changelog entries

* Ensure that UserLanguages telemetry initializer always runs after extensions are augmented
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by daniel-beck** [comment]
> The code is now similar to the ResourceDomainFilter:

What you're showing is just the code in this PR again, bad copy & paste?

**Review by daniel-beck — APPROVED** [review]
🤷 if it works (manual testing with custom start and end dates please).

**Comment by oleg-nenashev** [comment]
After several runs of initializer I actually got this error:

```
2020-05-05 09:11:14.017+0000 [id=25]    WARNING hudson.util.PluginServletFilter#addFilter: Filter instance is registered too early: jenkins.security.ResourceDomainFilter@524adee5
java.lang.Exception
        at hudson.util.PluginServletFilter.addFilter(PluginServletFilter.java:108)
```

So this fix is definitely not enough

**Review by MarkEWaite — APPROVED** [review]

**Comment by res0nance** [comment]
> After several runs of initializer I actually got this error:
> 
> ```
> 2020-05-05 09:11:14.017+0000 [id=25]    WARNING hudson.util.PluginServletFilter#addFilter: Filter instance is registered too early: jenkins.security.ResourceDomainFilter@524adee5
> java.lang.Exception
>         at hudson.util.PluginServletFilter.addFilter(PluginServletFilter.java:108)
> ```
> 
> So this fix is definitely not enough

I see that some filter adding operations do it after JOB_LOADED
https://github.com/jenkinsci/graphql-server-plugin/blob/ca276ee90ab5d0a5bdad07d4e013eff757112e2b/src/main/java/io/jenkins/plugins/graphql/filters/GraphQLCORSFilter.java#L31-L41

**Comment by oleg-nenashev** [comment]
> I see that some filter adding operations do it after JOB_LOADED

Userlanguages filter is not a problem, it can be enabled once the rest of the Jenkins starts up. There is no real need to enable it earlier. 

ResourceDomainFilter is a bit more tricky, because it is needed for showing the UI. It should be loaded as soon as possible before the instance starts processing the requests. I would not like to touch the initialization unless there is evidence that this issue happens outside Jenkinsfile Runner

**Comment by oleg-nenashev** [comment]
I fixed the JFR issue by https://github.com/jenkinsci/jenkinsfile-runner/pull/294 . As it was assumed above, it actually was not an issue on the Jenkins core side.

This fix still makes sense in general, but I will leave it to reviewers to decide whether we want to accept it as a general improvement

**Review by alecharp — APPROVED** [review]

**Review by fcojfernandez — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
I plan to merge it in 24 hours if no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information

**Comment by daniel-beck** [comment]
@oleg-nenashev Could you check classification and changelog message to make sure they're correct and useful to administrators? It doesn't look like there's much of a bugfix left here, for example.

**Comment by oleg-nenashev** [comment]
@daniel-beck for me it still looks a minor bug, but please feel free to convert it to internal if you prefer it

**Comment by daniel-beck** [comment]
@oleg-nenashev You wrote above:

> As it was assumed above, it actually was not an issue on the Jenkins core side.
> 
> This fix still makes sense in general, but I will leave it to reviewers to decide whether we want to accept it as a general improvement

So what bug does this change fix?

**Pull request merged**
